creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -1034,6 +1082,43 @@ class SnapbackSM {
       this.snapbackReconfigEnabled = this.nodeConfig.get('snapbackReconfigEnabled')
     }
   }
+
+  // Fetch clock values with max number of attempts
+  async fetchClockValues (secondary1, secondary2, wallet, maxClockFetchAttempts = 50) {
+    let secondariesToClockMap = {}
+    let attempts = 0
+
+    while (attempts++ < maxClockFetchAttempts) {
+      const [secondary1ClockResponse, secondary2ClockResponse] = await Promise.all(
+        [secondary1, secondary2].map(node => {
+          return async () => {
+            let clockValue = null
+            try {
+              clockValue = await CreatorNode.getClockValue(node, wallet, 5000)
+            } catch (e) { /* Swallow error */ }
+            return clockValue
+          }
+        })
+      )
+
+      if (secondary1ClockResponse) {
+        secondariesToClockMap[secondary1] = secondary1ClockResponse
+      }
+
+      if (secondary2ClockResponse) {
+        secondariesToClockMap[secondary2] = secondary2ClockResponse
+      }
+
+      if (secondariesToClockMap[secondary1] && secondariesToClockMap[secondary2]) {
+        return secondariesToClockMap
+      }
+    }
+
+    if (!secondariesToClockMap[secondary1]) { secondariesToClockMap[secondary1] = -1 }
```

Comment:
if we get into this scenario where one or both secondaries fails to return clock value after 50 attempts, rather than setting it to -1, i actually feel like we should just mark that secondary as unhealthy and cycle user off of it right?

maybe we can fetch these clock values at the beginning of issueUpdateReplicaSetOp, and on failure to retrieve we add that secondary to the unhelathyReplicasSet before calling determineNewReplicaSet

down to discuss

Reply:
discussed offline - good point, will add to unhealthyReplicaSet and let the big if / else case handle the reconfig as necessary